pasqal_cloud/__init__.py

```diff
@@ -251,28 +254,108 @@ def rebatch(
             job_ids: list of job ids to retry.
             status: status or list of statuses.
                 Will retry jobs currently matching this/these status/es.
-            min_runs: retry jobs with more or as many as this amount of runs.
-            mas_runs: retry jobs with less or as many as this amount of runs.
+            min_runs: retry jobs with more or as many as this number of runs.
+            max_runs: retry jobs with less or as many as this number of runs.
             start_date: retry jobs created at or after this datetime.
             end_date: retry jobs created at or before this datetime.
 
+        Returns:
+            Batch: the new batch made from the rebatch
+
         Raises:
             RebatchError if rebatch call failed.
         """
+        query_params: Dict[str, Any] = {}
+        if not isinstance(job_ids, EmptyFilter):
+            query_params["id"] = job_ids
+        if not isinstance(status, EmptyFilter):
+            query_params["status"] = status
+        if not isinstance(min_runs, EmptyFilter):
+            query_params["min_runs"] = min_runs
+        if not isinstance(max_runs, EmptyFilter):
+            query_params["max_runs"] = max_runs
+        if not isinstance(start_date, EmptyFilter):
+            query_params["start_date"] = start_date
+        if not isinstance(end_date, EmptyFilter):
+            query_params["end_date"] = end_date
+
+        params = RebatchFilterParams(**query_params)
         try:
-            new_batch_data = self._client.rebatch(
-                id,
-                job_ids=job_ids,
-                status=status,
-                min_runs=min_runs,
-                max_runs=max_runs,
-                start_date=start_date,
-                end_date=end_date,
+            new_batch_data = self._client._rebatch(
+                batch_id=id,
+                filters=params,
             )
         except HTTPError as e:
             raise RebatchError(e) from e
         return Batch(**new_batch_data, _client=self._client)
 
+    def get_jobs(
+        self,
+        job_ids: Union[List[UUID], EmptyFilter] = EmptyFilter(),
```

Comment:
Let's be lenient on the user inputs and accept both UUID and str. We need to make sure to cast it to a known type (str or UUID) when building the `JobFilterParams`